main.js

```diff
@@ -145,25 +155,24 @@ if (urlParams.downloadJAD) {
             dialog.parentElement.removeChild(dialog);
 
             jvm.addPath("midlet.jar", data.jarData);
-
-            fs.create("/midlet.jad", new Blob([ data.jadData ]), function() {
-              fs.create("/midlet.jar", new Blob([ data.jarData ]), function() {
-                resolve();
-              });
-            });
+            processJAD(data.jadData);
+
+            Promise.all([
+              new Promise(function(resolve, reject) {
+                fs.create("/midlet.jad", new Blob([ data.jadData ]), function() {
+                  resolve();
+                });
+              }),
+              new Promise(function(resolve, reject) {
+                fs.create("/midlet.jar", new Blob([ data.jarData ]), function() {
+                  resolve();
+                });
```

Comment:
These two anonymous functions that just call *resolve* could simply be references to *resolve* itself.